src/components/application_manager/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -72,6 +72,32 @@ mobile_apis::AppHMIType::eType StringToAppHMIType(const std::string &str) {
   }
 }
 
+std::string AppHMITypeToString(mobile_apis::AppHMIType::eType type) {
+  if (mobile_apis::AppHMIType::DEFAULT == type) {
+    return "DEFAULT";
+  } else if (mobile_apis::AppHMIType::COMMUNICATION == type) {
+    return "COMMUNICATION";
+  } else if (mobile_apis::AppHMIType::MEDIA == type) {
+    return "MEDIA";
+  } else if (mobile_apis::AppHMIType::MESSAGING == type) {
+    return "MESSAGING";
+  } else if (mobile_apis::AppHMIType::NAVIGATION == type) {
+    return "NAVIGATION";
+  } else if (mobile_apis::AppHMIType::INFORMATION == type) {
+    return "INFORMATION";
+  } else if (mobile_apis::AppHMIType::SOCIAL == type) {
+    return "SOCIAL";
+  } else if (mobile_apis::AppHMIType::BACKGROUND_PROCESS == type) {
+    return "BACKGROUND_PROCESS";
+  } else if (mobile_apis::AppHMIType::TESTING == type) {
+    return "TESTING";
+  } else if (mobile_apis::AppHMIType::SYSTEM == type) {
+    return "SYSTEM";
+  } else {
+    return NULL;
```

Comment:
@jacobkeeler This will cause core crash.  Probably we should have there some DCHECK or at lease log error